===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/auth2.c,v
retrieving revision 1.70
retrieving revision 1.71
diff -u -r1.70 -r1.71
--- src/usr.bin/ssh/auth2.c	2001/09/20 13:46:48	1.70
+++ src/usr.bin/ssh/auth2.c	2001/09/27 15:31:17	1.71
@@ -23,7 +23,7 @@
  */
 
 #include "includes.h"
-RCSID("$OpenBSD: auth2.c,v 1.70 2001/09/20 13:46:48 markus Exp $");
+RCSID("$OpenBSD: auth2.c,v 1.71 2001/09/27 15:31:17 markus Exp $");
 
 #include <openssl/evp.h>
 
@@ -118,7 +118,7 @@
 
 	x_authctxt = authctxt;		/*XXX*/
 
-	/* challenge-reponse is implemented via keyboard interactive */
+	/* challenge-response is implemented via keyboard interactive */
 	if (options.challenge_response_authentication)
 		options.kbd_interactive_authentication = 1;
 
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/auth2-chall.c,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- src/usr.bin/ssh/auth2-chall.c	2001/06/23 15:12:17	1.7
+++ src/usr.bin/ssh/auth2-chall.c	2001/09/27 15:31:17	1.8
@@ -23,7 +23,7 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 #include "includes.h"
-RCSID("$OpenBSD: auth2-chall.c,v 1.7 2001/06/23 15:12:17 itojun Exp $");
+RCSID("$OpenBSD: auth2-chall.c,v 1.8 2001/09/27 15:31:17 markus Exp $");
 
 #include "ssh2.h"
 #include "auth.h"
@@ -139,7 +139,7 @@
 }
 
 /*
- * try challenge-reponse, set authctxt->postponed if we have to
+ * try challenge-response, set authctxt->postponed if we have to
  * wait for the response.
  */
 int
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/compress.c,v
retrieving revision 1.14
retrieving revision 1.15
diff -u -r1.14 -r1.15
--- src/usr.bin/ssh/compress.c	2001/04/05 10:39:01	1.14
+++ src/usr.bin/ssh/compress.c	2001/09/27 11:58:16	1.15
@@ -12,7 +12,7 @@
  */
 
 #include "includes.h"
-RCSID("$OpenBSD: compress.c,v 1.14 2001/04/05 10:39:01 markus Exp $");
+RCSID("$OpenBSD: compress.c,v 1.15 2001/09/27 11:58:16 markus Exp $");
 
 #include "log.h"
 #include "buffer.h"
@@ -33,7 +33,7 @@
 buffer_compress_init_send(int level)
 {
 	if (compress_init_send_called == 1)
-		deflateEnd(&incoming_stream);
+		deflateEnd(&outgoing_stream);
 	compress_init_send_called = 1;
 	debug("Enabling compression at level %d.", level);
 	if (level < 1 || level > 9)
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/packet.c,v
retrieving revision 1.69
retrieving revision 1.70
diff -u -r1.69 -r1.70
--- src/usr.bin/ssh/packet.c	2001/06/25 08:25:38	1.69
+++ src/usr.bin/ssh/packet.c	2001/09/27 11:59:37	1.70
@@ -37,7 +37,7 @@
  */
 
 #include "includes.h"
-RCSID("$OpenBSD: packet.c,v 1.69 2001/06/25 08:25:38 markus Exp $");
+RCSID("$OpenBSD: packet.c,v 1.70 2001/09/27 11:59:37 markus Exp $");
 
 #include "xmalloc.h"
 #include "buffer.h"
@@ -1226,6 +1226,7 @@
 		log("packet_set_maxsize: bad size %d", s);
 		return -1;
 	}
+	called = 1;
 	debug("packet_set_maxsize: setting to %d", s);
 	max_packet_size = s;
 	return s;
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/scard.c,v
retrieving revision 1.14
retrieving revision 1.15
diff -u -r1.14 -r1.15
--- src/usr.bin/ssh/scard.c	2001/09/17 20:22:14	1.14
+++ src/usr.bin/ssh/scard.c	2001/09/28 09:49:31	1.15
@@ -24,7 +24,7 @@
 
 #include "includes.h"
 #ifdef SMARTCARD
-RCSID("$OpenBSD: scard.c,v 1.14 2001/09/17 20:22:14 markus Exp $"); 
+RCSID("$OpenBSD: scard.c,v 1.15 2001/09/28 09:49:31 djm Exp $");
 
 #include <openssl/engine.h>
 #include <sectok.h>
@@ -123,6 +123,7 @@
 	int len, sw, status = -1;
 
 	len = sw = 0;
+	n = NULL;
 
 	if (sc_fd < 0) {
 		status = sc_init();
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/ssh.c,v
retrieving revision 1.144
retrieving revision 1.145
diff -u -r1.144 -r1.145
--- src/usr.bin/ssh/ssh.c	2001/09/24 03:38:58	1.144
+++ src/usr.bin/ssh/ssh.c	2001/09/28 15:46:29	1.145
@@ -39,7 +39,7 @@
  */
 
 #include "includes.h"
-RCSID("$OpenBSD: ssh.c,v 1.144 2001/09/24 03:38:58 stevesk Exp $");
+RCSID("$OpenBSD: ssh.c,v 1.145 2001/09/28 15:46:29 markus Exp $");
 
 #include <openssl/evp.h>
 #include <openssl/err.h>
@@ -613,10 +613,10 @@
 	} else  {
 		snprintf(buf, sizeof buf, "%.100s/%.100s", pw->pw_dir,
 		    _PATH_SSH_USER_CONFFILE);
+		(void)read_config_file(buf, host, &options);
 
-		/* Read systemwide configuration file. */
+		/* Read systemwide configuration file after use config. */
 		(void)read_config_file(_PATH_HOST_CONFIG_FILE, host, &options);
-		(void)read_config_file(buf, host, &options);
 	}
 
 	/* Fill configuration defaults. */
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/sshconnect1.c,v
retrieving revision 1.39
retrieving revision 1.40
diff -u -r1.39 -r1.40
--- src/usr.bin/ssh/sshconnect1.c	2001/07/05 20:32:47	1.39
+++ src/usr.bin/ssh/sshconnect1.c	2001/09/27 15:31:17	1.40
@@ -13,7 +13,7 @@
  */
 
 #include "includes.h"
-RCSID("$OpenBSD: sshconnect1.c,v 1.39 2001/07/05 20:32:47 stevesk Exp $");
+RCSID("$OpenBSD: sshconnect1.c,v 1.40 2001/09/27 15:31:17 markus Exp $");
 
 #include <openssl/bn.h>
 #include <openssl/evp.h>
@@ -820,9 +820,9 @@
 	u_int clen;
 	char prompt[1024];
 	char *challenge, *response;
-	
-	debug("Doing challenge reponse authentication.");
-	
+
+	debug("Doing challenge response authentication.");
+
 	for (i = 0; i < options.number_of_password_prompts; i++) {
 		/* request a challenge */
 		packet_start(SSH_CMSG_AUTH_TIS);
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/sshd.8,v
retrieving revision 1.152
retrieving revision 1.153
diff -u -r1.152 -r1.153
--- src/usr.bin/ssh/sshd.8	2001/09/20 00:15:54	1.152
+++ src/usr.bin/ssh/sshd.8	2001/09/27 17:53:24	1.153
@@ -34,7 +34,7 @@
 .\" (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 .\" THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 .\"
-.\" $OpenBSD: sshd.8,v 1.152 2001/09/20 00:15:54 stevesk Exp $
+.\" $OpenBSD: sshd.8,v 1.153 2001/09/27 17:53:24 camield Exp $
 .Dd September 25, 1999
 .Dt SSHD 8
 .Os
@@ -1165,9 +1165,8 @@
 refused.
 The file should be world-readable.
 .It Pa /etc/hosts.allow, /etc/hosts.deny
-If compiled with
-.Sy LIBWRAP
-support, tcp-wrappers access controls may be defined here as described in
+Access controls that should be enforced by tcp-wrappers are defined here.
+Further details are described in
 .Xr hosts_access 5 .
 .It Pa $HOME/.rhosts
 This file contains host-username pairs, separated by a space, one per
===================================================================
RCS file: /usr/OpenBSD/cvs/src/usr.bin/ssh/ssh-keygen.c,v
retrieving revision 1.81
retrieving revision 1.82
diff -u -r1.81 -r1.82
--- src/usr.bin/ssh/ssh-keygen.c	2001/09/17 20:50:22	1.81
+++ src/usr.bin/ssh/ssh-keygen.c	2001/09/28 12:07:09	1.82
@@ -12,7 +12,7 @@
  */
 
 #include "includes.h"
-RCSID("$OpenBSD: ssh-keygen.c,v 1.81 2001/09/17 20:50:22 markus Exp $");
+RCSID("$OpenBSD: ssh-keygen.c,v 1.82 2001/09/28 12:07:09 djm Exp $");
 
 #include <openssl/evp.h>
 #include <openssl/pem.h>
@@ -491,6 +491,14 @@
 	status = 0;
 	log("loading key done");
 done:
+
+	memset(elements[0], '\0', BN_num_bytes(prv->rsa->q));
+	memset(elements[1], '\0', BN_num_bytes(prv->rsa->p));
+	memset(elements[2], '\0', BN_num_bytes(prv->rsa->iqmp));
+	memset(elements[3], '\0', BN_num_bytes(prv->rsa->dmq1));
+	memset(elements[4], '\0', BN_num_bytes(prv->rsa->dmp1));
+	memset(elements[5], '\0', BN_num_bytes(prv->rsa->n));
+
 	if (prv)
 		key_free(prv);
 	for (i = 0; i < NUM_RSA_KEY_ELEMENTS; i++)
